src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -254,14 +263,21 @@ impl DeviceManager {
             // convert the block driver to kata type
             VIRTIO_BLOCK_MMIO => KATA_MMIO_BLK_DEV_TYPE.to_string(),
             VIRTIO_BLOCK_PCI => KATA_BLK_DEV_TYPE.to_string(),
+            VM_ROOTFS_DRIVER_PMEM => VM_ROOTFS_DRIVER_PMEM.to_string(),
             _ => "".to_string(),
         };
-        block_config.driver_option = block_driver;
+        block_config.driver_option = block_driver.clone();
 
-        // generate virt path
-        if let Some(virt_path) = self.get_dev_virt_path(DEVICE_TYPE_BLOCK)? {
-            block_config.index = virt_path.0;
-            block_config.virt_path = virt_path.1;
+        if block_driver == *VM_ROOTFS_DRIVER_PMEM {
+            block_config.index = self.shared_info.get_and_set_pmem_index();
+            block_config.virt_path = format!("pmem{}", block_config.index);
+        } else {
+            // generate block device index and virt path
+            // safe here, Block device always has virt_path.
+            if let Some(virt_path) = self.get_dev_virt_path(DEVICE_TYPE_BLOCK)? {
+                block_config.index = virt_path.0;
+                block_config.virt_path = virt_path.1;
+            }
```

Comment:
For case PMEM, a special block device case, we should handle it in a generic way as much as possible. And the index and virt_path are handled in the  get_dev_virt_path is better or maybe some other solutions.

Reply:
fixed